src/main/java/dev/snowdrop/release/services/CPaaSConfigUpdateService.java

```diff
@@ -0,0 +1,111 @@
+                Stream.Builder<File> builder = Stream.builder();
+                fileStream.forEach(file -> {
+                    builder.add(file);
+                });
+                builder.add(advisoryFile);
+                fileStream = builder.build();
+            }
+            return fileStream;
+        });
+    }
+
+    public Stream<File> updateCPaaSFiles(String release, File repo, String previousRelease, boolean isErCr, boolean isSecurityAdvisory) {
+        String[] releaseArray = release.split("\\.");
+        if (releaseArray.length != 3) {
+            throw new IllegalArgumentException("Invalid release: " + release
+                + ". Must follow major.minor.fix format (e.g. 2.4.3).");
+        }
+        final String repoPath = repo.getAbsolutePath();
+        List<File> fileList = new ArrayList(3);
+        final Path productPath = Paths.get(String.format(repoPath + "/product.yml"));
+        final File productFile = productPath.toFile();
+        final Path releasePath = Paths.get(String.format(repoPath + "/release.yml"));
+        final File releaseFile = releasePath.toFile();
+        if (productFile.exists()) {
+            try {
+                InputStream productIS = new FileInputStream(productFile);
+                CPaaSProductFile cpaasProductFile = factory.createCPaaSProductFromStream(productIS);
+                CPaaSProduct cpaasProduct = cpaasProductFile.getProduct();
+                cpaasProduct.getRelease().setVersion(release);
+                CPaaSProject projects = cpaasProduct.getProjects().get(0);
+                if ("spring-boot-parent".equalsIgnoreCase(projects.getName())) {
+                    CPaaSComponent sbComponent = projects.getComponents().get(0);
+                    if ("spring-boot".equalsIgnoreCase(sbComponent.getName())) {
+                        CPaaSBuild sbBuild = sbComponent.getBuilds().get(0);
+                        if ("spring-boot".equalsIgnoreCase(sbBuild.getName())) {
+                            sbBuild.getPigSource().setRoot(String.format("spring-boot/%s.%s", releaseArray[0], releaseArray[1]));
+                        }
+                    }
+                }
+                factory.saveTo(cpaasProductFile, productFile);
+                fileList.add(productFile);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        try {
+            ReleaseMustache releaseMustache = new ReleaseMustache(isErCr, isSecurityAdvisory, null, release, previousRelease);
+            StringWriter writer = new StringWriter();
+            Utility.mf.compile(RELEASE_TEMPLATE).execute(writer, releaseMustache).flush();
+            CPaaSReleaseFile cpaasReleaseFile = factory.createCPaaSReleaseFromStream(new ByteArrayInputStream(writer.toString().getBytes()));
+            factory.saveTo(cpaasReleaseFile, releaseFile);
+            fileList.add(releaseFile);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
We should probably do something about this exception like fail the release…

Reply:
raised a runtime exception